plugins/faustwp/includes/replacement/previewlinks.js

```diff
@@ -2,54 +2,63 @@
  * XXX: Please remove this once this issue is resolved: https://github.com/WordPress/gutenberg/issues/13998
  */
 
-window.addEventListener('DOMContentLoaded', function () {
-  jQuery(document).ready(function () {
-    // Get the correct preview link via wp_localize_script
-    const previewLink = window._faustwp_preview_link
-      ? window._faustwp_preview_link._preview_link
-      : undefined;
-
-    /**
-     * Check to make sure there is a preview link before continuing, as there may not be a preview link
-     * for every instance the block editor is enqueued (e.g. /wp-admin/widgets.php)
-     */
-    if (!previewLink) {
-      return;
+document.addEventListener('DOMContentLoaded', function() {
+  // Get the preview data via wp_localize_script
+  const faustPreviewData = window._faustwp_preview_data;
+
+  /**
+   * Check to make sure there is a preview link before continuing, as there may not be a preview link
+   * for every instance the block editor is enqueued (e.g. /wp-admin/widgets.php)
+   */
+  if (!faustPreviewData) {
+    return;
+  }
+
+  const wpVersion = faustPreviewData._wp_version;
+  const faustPreviewLink = faustPreviewData._preview_link;
+
+  function debounce(func, wait) {
+    let timeout;
+    return function() {
+      const context = this;
+      const args = arguments;
+      clearTimeout(timeout);
+      timeout = setTimeout(function() {
+        func.apply(context, args);
+      }, wait);
+    };
+  }
+
+  // Handle potential breaking changes from WordPress.
+  function getPreviewLinksByVersion(version) {
+    switch (version) {
+      default:
+        return {
+          headerLink: document.querySelector('.edit-post-header-preview__grouping-external a'),
```

Comment:
This query selector can only be used once since we replace the node in line 49 thus making it problematic. If we attempt to read this from a deleted node we may find issues.